### DISCUSSION TITLE
how to use html or java in cosmos

### DISCUSSION THREAD

**Discussion by EndermanPC** [CosmosOS/Cosmos#2005] [Q&A]
i want use html or java in cosmos, how?

**Comment by reisxd** [comment]
You can't

**Reply by EndermanPC** [reply]
why, can me use c++?

**Reply by reisxd** [reply]
No, only C#

**Reply by EndermanPC** [reply]
hmm, and how me make a test editon in c#?

**Reply by EndermanPC** [reply]
this, [https://codereview.stackexchange.com/questions/191503/simple-console-text-editor](url)

**Reply by zarlo** [reply]
you can use basic c++ programs (you should not use c++ in cosmos right now but) but you will have to make a ELF loader and the programs will have full access to memory

**Reply by EndermanPC** [reply]
why i should not use c++ in cosmos right now
